commands/command_pictures.cpp

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: GPL-2.0
+
+#include "command_pictures.h"
+#include "core/subsurface-qt/divelistnotifier.h"
+
+namespace Command {
+
+static picture *dive_get_picture(const dive *d, const QString &fn)
```

Comment:
> this isn't multithreaded so we know the pointer will remain valid? I.e. we know that while we keep this pointer the table never expands?

Exactly. Since this isn't run inside a critical section, neither pointers nor indexes would be safe in a multi-threaded context. If you look at the callers, the pointer is only used for an access. Indeed, the whole point of the function is to access the picture via its index (which in this case is its filename).